### PR TITLE
Change Account Email

### LINKED ISSUE #113: Change Main Contact Email
[Project/Issue Tracker](https://github.com/Code-4-Community/lucys-love-bus-frontend/projects/1)

### Objective

As a user I would like to be able to change my email!
 
### Tech Spec
- [x] A specific page for changing email
- [x] Page linked from authenticated navbar
- [x] uses POST /user/change_email

### Design Artifacts

Provide a list of design artifacts on which the above tech spec is based. This could be an external link to an artifact the Design or UX team provided. Include authors names so that the task owner can reach out.

### Notes

Any clarifying content unrelated to the above items (Or, just a word of encouragement, like "You're doing great!")

### PR DESCRIPTION
**Pull request by anirudh723**
## Issue

Closes #113 

## Changes

- Added a change account email page
- Added the API route to make the request to, and the page route to host the new page
- Connected the page to the new page route, and to the navbar dropdown
- Added the request making function to `protectedApiClient`

## Screenshots
Change Account Email page:
image

Error message when entering the same email as my current one:
image

Error message when entering the wrong password:
image

Confirmation message after successfully changing email:
image

## Verification Steps
- Tested the two exceptions being thrown by the backend by entering the wrong password, and entering the same email as my current one. 
- Made sure that the confirmation page shows up after successfully changing my email.
- Refreshed the page, and made sure it works, for ex: trying to change my email from the new email I just made to the same new one, which properly displayed the error message.

### CODE REVIEW

File: src/components/ConfirmationMessage.tsx
Comment (rymaju) [inline src/components/ConfirmationMessage.tsx:24]: Hmm I'm not sure if I'm happy with this change but it depends how you use it later.

File: src/containers/changeAccountEmail/index.tsx
Comment (rymaju) [inline src/containers/changeAccountEmail/index.tsx:17]: Lets type useState
```suggestion
  const [loading, setIsLoading] = useState<boolean>(false);
```

File: src/containers/changeAccountEmail/index.tsx
Comment (rymaju) [inline src/containers/changeAccountEmail/index.tsx:21]: We should create an interface for the form types, it will be the same interface as the request for the API so we can also reuse it there
```suggestion
  const onFinishChangeEmail = async (values: ChangeEmailRequest) => {
```

```
interface ChangeEmailRequest {
    newEmail: string;
    password: string;
}
```

File: src/containers/changeAccountEmail/index.tsx
Comment (rymaju) [inline src/containers/changeAccountEmail/index.tsx:30]: Awesome🎉

File: src/containers/changeAccountEmail/index.tsx
Comment (rymaju) [inline src/containers/changeAccountEmail/index.tsx:39]: Oh this isnt how meta tags work, the name is for the label of the meta tag and it has to be one of a few specific strings
```suggestion
          name="description"
```

**Review by rymaju — CHANGES_REQUESTED** [review]
A few very minor things that I want fixed before we merge this in. Other than that though great work! 🥳

File: src/components/ConfirmationMessage.tsx
Comment (anirudh723) [reply src/components/ConfirmationMessage.tsx:24]: Some of the confirmation message pages may have a "click here to go back to the homepage" or something, so the type flexibility makes it easier to do that

**Review by rymaju — APPROVED** [review]
Looks good 👍 

I would wait on another approval, but otherwise feel free to merge this in whenever

File: src/components/ConfirmationMessage.tsx
Comment (jackblanc) [reply src/components/ConfirmationMessage.tsx:24]: Seems reasonable to me

File: src/components/navbar/index.tsx
Comment (jackblanc) [inline src/components/navbar/index.tsx:105]: We should probably just delete menu items that don't go anywhere

File: src/containers/changeAccountEmail/index.tsx
Comment (jackblanc) [inline src/containers/changeAccountEmail/index.tsx:33]:
```suggestion
  const [changeEmail, setChangeEmail] = useState<AsyncRequest<void, any>>(AsyncRequestNotStarted());

  const onFinishChangeEmail = async (values: ChangeEmailRequest) => {
    try {
      setChangeEmail(AsyncRequestLoading());
      await ProtectedApiClient.changeAccountEmail(values);
      setChangeEmail(AsyncRequestCompleted());
    } catch (err) {
      setChangeEmail(AsyncRequestFailed(err));
    } 
  };
```

Personally, I think this is a lot more readable.

**Review by jackblanc — COMMENTED** [review]
I haven't talked much before about using `AsyncRequest` outside of Redux, but I think it could be pretty valuable in this case. See my comment below and let me know what you think!

File: src/containers/changeAccountEmail/index.tsx
Comment (rymaju) [reply src/containers/changeAccountEmail/index.tsx:33]: Yes!!! This should be in the docs! 😍

File: src/components/navbar/index.tsx
Comment (anirudh723) [reply src/components/navbar/index.tsx:105]: I think the goal is to create what we have in the Vue version - just have Settings item in the navbar, and the settings page has all the buttons to the individual setting pages (change email, change password, edit family info, etc). So in short, yes these will go away

File: src/containers/changeAccountEmail/index.tsx
Comment (anirudh723) [reply src/containers/changeAccountEmail/index.tsx:33]: Looks dope

**Review by onb6 — APPROVED** [review]
this looks really good!

File: src/containers/changeAccountEmail/index.tsx
Comment (anirudh723) [reply src/containers/changeAccountEmail/index.tsx:33]: @jackblanc What should I pass into `AsyncRequestCompleted()` on line 24 of the suggested change?

Comment (anirudh723) [reply src/containers/changeAccountEmail/index.tsx:33]: Just gonna pass it in undefined

**Pull request merged**
